htcanalyze/resource.py

```diff
@@ -0,0 +1,133 @@
+"""Class to define HTCondor Joblog resources."""
+
+from typing import List
+from numpy import nan_to_num as ntn
+
+
+class Resource:
+    """
+    Class of one single HTCondor-JobLog resource.
+
+    One file contains usually multiple resources,
+    so that nested lists represent a collection of job resources
+    """
+
+    level_colors = {'error': 'red', 'warning': 'yellow',
+                    'light_warning': 'yellow2', 'normal': 'green'}
+
+    def __init__(self,
+                 name: str,
+                 usage: float,
+                 requested: float,
+                 allocated: float,
+                 warning_level: str = None):
+        # naming in __init__ arguments should be
+        # consistent with names in the resources dictionary
+        self.name = name
+        self.usage = usage
+        self.requested = requested
+        self.allocated = allocated
+        self.warning_level = warning_level
```

Comment:
Since `warning_level` is not passed on by htcondor, you don't need it in the init args. Just do `self.warning_level = None` here.

Reply:
Well what if this class is used again in future. I usually like to have these kind of constructors, whenever I want to extend  the script. Also I usually think of other developers that might want to use this code in future. Someone might be happy about this. But in this case we can just be selfish here :) 

But I hope you get my point, that even if something is not especially used, it might be useful in other circumstances.

Reply:
Don't worry about "what if". Unless you have a good reason to assume something is going to be used in a certain way later, don't bother implementing it. Otherwise your code will be full of small idiosyncrasies that may or may not ever be used in the future.

That said, you can of course do 

```
foo = Resource('foo', 'bar', ...)
foo.warning_level = 'error'
```